### ISSUE TITLE
Support Zstd-compressed ELF debug information sections

### ISSUE THREAD

**Issue by XVilka** [rizinorg/rizin#3019]
The support was introduced recently (not yet released) LLVM:
- https://github.com/llvm/llvm-project/commit/5e0464e38bcb63186e5c5eaafe550e92143cab43
- https://github.com/llvm/llvm-project/commit/449f2ca146dde0ef14c9837f71d4a0be92ca4843

And GCC:
- https://gcc.gnu.org/git/?p=gcc.git;a=commitdiff;h=b7723e135334df95597c8c744276b9de5a88214a

More technical details are available in this MaskRay's blog post: https://maskray.me/blog/2022-09-09-zstd-compressed-debug-sections

Basically, it adds a new `ch_type` for `ELFCOMPRESS_ZSTD  2`

**Comment by ret2libc** [comment]
I think we should be able to add support for this without additional dependencies.

**Comment by XVilka** [comment]
Binutils 2.40 support it too now: https://lists.gnu.org/archive/html/info-gnu/2023-01/msg00003.html

**Comment by XVilka** [comment]
MaskRay proposed the support for compressing any arbitrary section: https://discourse.llvm.org/t/rfc-compress-arbitrary-sections-with-ld-lld-compress-sections/71674

Thus, the design of this feature should keep this in mind to be future-proof.
